src/Pate/Equivalence.hs

```diff
@@ -468,6 +564,9 @@ getRegionEquality sym memEq memO memP = case memEq of
   MemEqAtRegion stackRegion -> MT.memEqAtRegion sym stackRegion memO memP
   MemEqOutsideRegion stackRegion -> MT.memEqOutsideRegion sym stackRegion memO memP
 
+newtype RegisterCondition sym arch =
```

Comment:
Can you document a bit about what this is meant to be used for?

What do the predicates in the `RegState` denote?